src/algebra/group/units.lean

```diff
@@ -333,10 +333,13 @@ by cases ha with a ha; rw [←ha, units.mul_left_inj]
 
 /-- The element of the group of units, corresponding to an element of a monoid which is a unit. -/
 noncomputable def is_unit.unit [monoid M] {a : M} (h : is_unit a) : units M :=
-classical.some h
+{ val := a,
+  inv := (classical.some h).inv,
+  val_inv := by { convert (classical.some h).val_inv, exact (classical.some_spec h).symm },
+  inv_val := by { convert (classical.some h).inv_val, exact (classical.some_spec h).symm } }
```

Comment:
Shorter as
```suggestion
(classical.some h).copy a (classical.some_spec h).symm _ rfl
```

Reply:
Not sure this really improves readability :-)

Reply:
I'd argue it does, it reads as "this is just `classical.some` but definitionally better. It also provides a nice example for how to use `copy`, which is designed for precisely this type of use.

Reply:
Not completely convinced (you need to think a bit to see what this is doing, while for the other one it's obvious), but let's merge it anyway if you like it.